ui/src/wallet.ts

```diff
@@ -0,0 +1,135 @@
+import WalletConnect from "@walletconnect/client";
+import { convertUtf8ToHex } from "@walletconnect/utils";
+import QRCodeModal from "@walletconnect/qrcode-modal";
+import WalletConnectWeb3Provider from "@walletconnect/web3-provider";
+import Web3 from "web3";
+import { writable, Readable } from "svelte/store";
+
+export enum Status {
+  Connected = "CONNECTED",
+  NotConnected = "NOT_CONNECTED",
+}
+
+type State =
+  | { status: Status.NotConnected }
+  | { status: Status.Connected; connected: Connected };
```

Comment:
There is a lot of connecting in this pr. :laughing: It would be helpful to give at leas this field and type a different name.

Reply:
do you have a suggestion? thought the same but name it `Wallet` doesn't help and could come up with a better name.